### DISCUSSION TITLE
Wait until "domcontentloaded" event is fired?

### DISCUSSION THREAD

**Discussion by waynehoover** [rubycdp/ferrum#378] [General]
I would like to do this: `wait_for_idle(events: ['domcontentloaded'])`

In puppeteer we can do `waitForNetworkIdle({waitUntil: ['domcontentloaded']})` 

Is it possible to wait until some event is emitted in JS in ferrum?

**Comment by route** [comment]
Ferrum returns back to you when the page you visited with `go_to` is ready to work with and hence `domcontentloaded` already fired (in CDP it's called `Page.domContentEventFired`) so no need to wait for such event.
